keyboards/keychron/q1/q1.c

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2021 @ Keychron (https://www.keychron.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "q1.h"
+
+
+void set_target_os(bool layer)
+{
+	if(layer)
+	{
+		default_layer_set(1UL << 0);
+	}
+	else
+	{
+		default_layer_set(1UL << 2);
+	}
+}
+
+void dip_switch_update_kb(uint8_t index, bool active) 
+{ 
+    if(index == 0)
+	{
+		set_target_os(!active);
+    }
+}
```

Comment:
You could just do this: 
```suggestion
void dip_switch_update_kb(uint8_t index, bool active) { 
    if (index == 0) {
        default_layer_set(1UL << (active ? 2 : 0)); 
    }
    dip_switch_update_user(index, active);
}
```